ironfish-cli/src/trusted-setup/server.ts

```diff
@@ -329,7 +345,8 @@ export class CeremonyServer {
     const hash = await verifyTransform(oldParamsDownloadPath, newParamsDownloadPath)
 
     client.logger.info(`Uploading verified contribution`)
-    const destFile = 'params_' + nextParamNumber.toString().padStart(4, '0')
+    const nameSuffix = client.name === undefined ? '' : `_${client.name.split(' ').join('_')}`
+    const destFile = 'params_' + nextParamNumber.toString().padStart(5, '0') + nameSuffix
```

Comment:
Paired with the comment above, if you remove restrictions from inputted text, you probably want to not use that text for file system naming. I would suggest just always using the anon path. Is there a value add in having the user name in the file name?

Reply:
We don't have any database beyond S3 for the trusted setup process (or rather, we're treating the S3 key-value store as our database). So the most immediately obvious place to store the contributor's name is in the filename, which is limited to support for these characters: https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html

Other options I can think of:

- Add the contributor to the file's metadata. It's a bit more inconvenient to retrieve, but S3 apparently allows arbitrary metadata values on files: https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html
- Create a separate file with metadata for the contribution. Might be a bit more straightforward than using metadata since we already have support for uploading files, but does raise concurrency issues (if upload of params succeeds but metadata fails, or vice versa, how do we handle that error state?)

Generally though, I think this is pretty low priority. If one of the above are easy to implement, feel free, but I don't think this should block. E.g. [Aleo](https://setup.aleo.org/) and [ZCash](https://github.com/zcash-hackworks/sapling-mpc/wiki) didn't support complex unicode characters

Reply:
Yea there is no value in having the name in the file name. This was more a matter of just finding a way to store a name in general like Derek mentioned. This wasn't in our original spec and isn't necessary per se. But I was thinking that if we did want to have some display of contributors later on, it wouldn't be possible to retroactively get peoples' names. Really just a simple name would do the job of identification. I understand the concern that we may be making the overall experience slightly worse by asking them to enter a name using a very restrictive character set. I'm going to look into S3 metadata quickly to see if that would be easy to implement